core/src/main/java/at/compus02/swd/ss2022/game/factories/PlayerFactory.java

```
@@ -0,0 +1,47 @@
+package at.compus02.swd.ss2022.game.factories;
+
+import at.compus02.swd.ss2022.game.gameobjects.*;
+
+public class PlayerFactory extends Factory {
+
+    private static PlayerFactory instance;
+
+    public PlayerFactory() {
+    }
+
+    public static PlayerFactory getInstance() {
+        if (instance == null) {
+            instance = new PlayerFactory();
+        }
+        return instance;
+    }
+
+    @Override
+    public Player create(GameObjectType type) {
+
+        Player player = null;
+
+        if (type == GameObjectType.PLAYER) {
+            player = new Player();
+        } else {
+            System.out.println("Wrong GameObjectType");
```

in dem fall wäre das schon eine exception wert, frei nach dem princip "fail early, not silent". sonst fallen manche fehler ewig nicht auf